torch/optim/optimizer.py

```diff
@@ -76,6 +76,27 @@ def _default_to_fused_or_foreach(params: List[torch.Tensor],
     return fused, foreach
 
 
+def _warn_step_no_param_with_grad(stacklevel=2):
+
+    def is_first_time():
+        if not hasattr(_warn_step_no_param_with_grad, 'has_warned'):
+            return True
+        else:
+            return not _warn_step_no_param_with_grad.__dict__['has_warned']
```

Comment:
```suggestion
        return not hasattr(_warn_step_no_param_with_grad, 'has_warned') or not _warn_step_no_param_with_grad.__dict__['has_warned']:
```
Could DeMorgan's Law it for further effect but this may be clearer.